pkg/pillar/types/errortime.go

```diff
@@ -34,36 +110,48 @@ func (etPtr *ErrorAndTime) SetError(errStr string, errorTime time.Time) {
 	}
 	etPtr.Error = errStr
 	etPtr.ErrorTime = errorTime
+	etPtr.ErrorSeverity = ErrorSeverityError
+	etPtr.ErrorEntities = []*ErrorEntity{}
+	etPtr.ErrorRetryCondition = ""
+}
+
+// SetErrorWithDescription is when time and description are specified
+// it sets ErrorSeverityError in case of unspecified ErrorSeverity
+// it sets ErrorTime to time.Now() in case of no time provided
+func (etPtr *ErrorAndTime) SetErrorWithDescription(errDescription ErrorDescription) {
+	if errDescription.Error == "" {
+		logrus.Fatal("Missing error string")
+	}
+	etPtr.ErrorDescription = errDescription
+	if etPtr.ErrorSeverity == ErrorSeverityUnspecified {
+		etPtr.ErrorSeverity = ErrorSeverityError
+	}
+	etPtr.ErrorTime = errDescription.ErrorTime
```

Comment:
One question which we should think about (thus no need to change the code in this PR) is whether we want to update the ErrorTime each time we retry and get the same error.
If we don't we could take a couple of different approaches:
1. Hide it in this function; if errorDescription is the same then don't change errorTime.
2. Add some different API (UpdateErrorWithDescription)?

Keeping the timestamp of the first time we saw the error might help with determining the severity. However, it could be that some agents clear the error as part of kicking off a retry... 

Needs some more thought and investigation once we're done with this PR.